app/components/HighlighterModule/highlighterModule.js

```diff
@@ -0,0 +1,437 @@
+        for(index=0;index<activeTopics.length;index+=1) {
+          if (i.topic == topic_list[index]) {
+            activeTopics[index] = active_state
+          }
+        }
+        // (6) Activate/Deactivate Sources
+        if (active_state){
+          var active = {start: i.source.start, end: i.source.end, text: i.source.text, top: i.source.topic};
+          activeSources = activeSources.concat([active]);
+        } else {
+          var active = {start: i.source.start, end: i.source.end, text: i.source.text, top: i.source.topic};
+          var source_index = -1;
+          var index = 0;
+          if (activeSources){
+            while (index < activeSources.length) {
+              var s = activeSources[index];
+              if (s.start == active.start && s.end == active.end) {
+                source_index = index;
+                break;
+              }
+              index += 1;
+            }
+          }
+          activeSources.splice(source_index, 1);
+        }
+        temp_index += 1;
+      }
+      return final;
+    }
+
+    function mergeHighlights(list) {
+      console.log('merge highlights')
+      var newlist = [];
+      var n = list.length;
+      for (var i = 0; i < n; i++) {
+        var newrange = Object.assign({}, list[i]);
+        var start = list[i].start;
+        var end = list[i].end;
+        var j = i + 1;
+        var check = true;
+        while (j < n) {
+          if (list[i].topic === list[j].topic) {
+            if (list[j].start < end) {
+              end = Math.max(end, list[j].end)
+              list.splice(j,1);
+              newrange.end = end;
+              newrange.text = privateText.substring(start, end)
+              n-=1;
+            } else {
+              break;
+            }
+          } else {
+            j++;
+          }
+        }
+        newlist.push(newrange);
+      }
+      return newlist;
+    }
+
+    function getOffset(childNodes, targetNode) {
+      var offset = 0;
+      for (var i in childNodes) {
+        var childNode = childNodes[i];
+        if (childNode === targetNode) {
+          break;
+        } else {
+          if (childNode.textContent == null) {
+            break;
+          } else {
+            offset += childNode.textContent.length;
+          }
+        }
+      }
+      return offset;
+    }
+
+    /* ----------- Private Listener Methods ----------- */
+
+    function handleKeydown(e) {
+      console.log('handleKeyDown');
+      e.preventDefault();
+      var code = e.keyCode;
+      if (code == 8 || code == 46) {
+        if (privateSelected.length > 0) {
+          deleteHighlight(privateSelected);
+        }
+      } else if (47 < code < 58) { //Temporary section for updating currenttopic
+        code -= 48;
+        privateCurrentTopic = code;
+      }
+    }
+
+    function handleMouseup(e) {
+      console.log('handleclick');
+
+      var container = document.getElementById('highlighter');
```

Comment:
If you're referring to the codes 47 to 58 that was for changing topics with the 0-9 keys (corresponding to the order of the topics).